[cmd] Parse print review status

csordasmarton opened the pull request:
> Closes #2596

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

gyorb commented on docs/analyzer/user_guide.md:
```diff
@@ -1096,7 +1101,10 @@ optional arguments:
+                        longest match will be removed.
+  --review-status [REVIEW_STATUS [REVIEW_STATUS ...]]
+                        Filter results by review statuses. (default:
+                        ['confirmed', 'unreviewed'])
```
Please list the other valid values here too.

gyorb commented on analyzer/codechecker_analyzer/cmd/check.py:
```diff
@@ -567,6 +567,13 @@ def add_arguments_to_parser(parser):
+                        dest="review_status",
+                        metavar='REVIEW_STATUS',
+                        default=['confirmed', 'unreviewed'],
+                        help="Filter results by review statuses.")
```
Please list all the valid filter values here.

csordasmarton commented:
@gyorb I fixed it. I have one more question. If there is not source code comment for a bug, do we want to show the default `Unreviewed` review status on parse?

E.g:
```sh
[HIGH] source_code_comments.cpp:7:3: suspicious usage of 'sizeof(K)'; did you mean 'K'? [bugprone-sizeof-expression] [Unreviewed]
```

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- analyzer/codechecker_analyzer/cmd/parse.py  1
- analyzer/codechecker_analyzer/cmd/check.py  1
         

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

gyorb commented:
That is a good question. Showing the unreviewed state would be more in sync with the database, but probably it would add a lot of noise to the output. By default if there is no review comment the definition is that the report is unreviewed.
Probably I would not add it first only if it is requested, but it should be shown up in the json output.
@bruntib @gamesh411  what do you think about printing the Unreviewed status for each report without any review comment, would that be too much noise in the output?

csordasmarton commented:
@gyorb As we discussed it today we do not want to show the Unreviewed status for reports without any review comment.

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- analyzer/codechecker_analyzer/cmd/parse.py  1
- analyzer/codechecker_analyzer/cmd/check.py  1
         

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

csordasmarton commented:
@gyorb On the ui by default we set this filter to unreviewed and confirmed. To be more consistent with this I set this filter on the command line also to show only unreviewed and confirmed bugs.

I fixed your other comments and I added a couple of new test cases (empty review-status filter and show only confirmed bug).

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- analyzer/codechecker_analyzer/cmd/parse.py  1
- analyzer/codechecker_analyzer/cmd/check.py  1
         

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- analyzer/codechecker_analyzer/cmd/check.py  1
- analyzer/codechecker_analyzer/cmd/parse.py  1
         

Clones removed
==============
+ web/server/tests/unit/test_source_code_comment.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5187760&bid=17020240)
